adapters/handlers/grpc/prepare_reply.go

```diff
@@ -60,27 +68,60 @@ func searchResultsToProto(res []any, start time.Time, searchParams dto.GetParams
 	return out, nil
 }
 
-func extractAdditionalProps(asMap map[string]any, additionalPropsParams additional.Properties) (*pb.ResultAdditionalProps, error) {
+func extractAdditionalProps(asMap map[string]any, additionalPropsParams additional.Properties, firstObject bool) (*pb.ResultAdditionalProps, string, error) {
 	err := errors.New("could not extract additional prop")
+	_, generativeSearchEnabled := additionalPropsParams.ModuleParams["generate"]
+
 	additionalProps := &pb.ResultAdditionalProps{}
-	if additionalPropsParams.ID {
+	if additionalPropsParams.ID && !generativeSearchEnabled {
 		idRaw, ok := asMap["id"]
 		if !ok {
-			return nil, err
+			return nil, "", errors.Wrap(err, "get id")
 		}
 
 		idStrfmt, ok := idRaw.(strfmt.UUID)
 		if !ok {
-			return nil, err
+			return nil, "", errors.Wrap(err, "format id")
 		}
 		additionalProps.Id = idStrfmt.String()
 	}
 	_, ok := asMap["_additional"]
 	if !ok {
-		return additionalProps, nil
+		return additionalProps, "", nil
 	}
 
 	additionalPropertiesMap := asMap["_additional"].(map[string]interface{})
+	generativeGroupResults := ""
+	// id is part of the _additional map in case of generative search - don't aks me why
+	if generativeSearchEnabled {
+		idRaw, ok := additionalPropertiesMap["id"]
+		if !ok {
+			return nil, generativeGroupResults, errors.Wrap(err, "get id generative")
```

Comment:
although `generativeGroupResults` is empty at this point I would explicitly return `""` for clarity